enf-reducer/gather-junctors.metta

```diff
@@ -1,48 +1,29 @@
-(: gatherJunctors (-> Tree Tree Tree))
-(= (gatherJunctors $currentNode $centerNode)
-      (let* (
-              ($currentNodeType (getNodeType $currentNode))
-              ($centerNodeType (getNodeType $centerNode))
-              ($currentLeft (getChild $currentNode L))
-              ($currentRight (getChild $currentNode R))
-              ((TreeNode (Value $valueName $constraint $currentNodeType) $guardSet $children) $currentNode)
+; a function to merge two tuples (A B C) (D E F) ==> (A B C D E F)
+(= (concat-tuple $xs $ys)
+    (if (== $xs ()) $ys
+        (let* (($head (car-atom $xs))  
+               ($tail (cdr-atom $xs)) 
+               ($tail-new (concat-tuple $tail $ys))  
+              )
+          (cons-atom $head $tail-new)  
+        )
+    )
+)
+
+(: gather-junctors (-> $expr $prev $expr))
+(= (gather-junctors $expr $prev)
+    ( case $expr
+        (
+            (($op $a $b)
+                ( if (== $prev ())

```

Comment:
I think you can remove this part of the code and it will still work as expected

Reply:
That condition is required to handle the case where the output is enclosed in double braces, like (( )).

it will be passed with the test case.  $prev will be **( )** for the _root node_